config/config.py

```diff
@@ -672,7 +672,7 @@ def onemkl_library():
             + '-I' + root + '/linux/include/sycl '
     env = {'LIBS': libs,
            'CXXFLAGS': inc + define('HAVE_ONEMKL')
-           + ' -fsycl -Wno-deprecated-declarations -fhonor-nan-compares'}
```

Comment:
Don't you need space before `-fsycl`? Otherwise it will be `-DLAPACKPP_HAVE_ONEMKL-fsycl` (no spaces).

Are the `-Wno-deprecated-declarations -fhonor-nan-compares` no longer needed?

Reply:
"Don't you need space before -fsycl? Otherwise it will be -DLAPACKPP_HAVE_ONEMKL-fsycl (no spaces)."
It comes out fine in my build, but maybe that's because the define is putting the value in include/lapack/defines.h.
I will add the space back, just to play it safe.

"Don't you need space before -fsycl? Otherwise it will be -DLAPACKPP_HAVE_ONEMKL-fsycl (no spaces).
Are the -Wno-deprecated-declarations -fhonor-nan-compares no longer needed?"

The deprecated-declarations problem went away ... perhaps Intel updated some headers.
The nan-compares addresses the test/matrix-generation code, and I use the fp-model=precise flag instead.

Reply:
After `make config gpu_backend=onemkl`, look in config/log.txt:
```
----------------------------------------
SYCL and oneMKL libraries
--------------------
    -lmkl_sycl -lsycl -lOpenCL
>>> icpx -std=c++17 -O2 -MMD -Wall -Wno-unused-local-typedefs -Wno-unused-function -fopenmp  -DLAPACK_FORTRAN_ADD_  -DLAPACK_COMPLEX_RETURN_ARGUMENT -DLAPACK_HAVE_MKL -DLAPACK_HAVE_CBLAS -DLAPACK_HAVE_LAPACK -DLAPACK_VERSION=31001 -DLAPACK_HAVE_XBLAS -DLAPACK_HAVE_LAPACKE -I/apps/spacks/2023-05-19/opt/spack/linux-rocky9-x86_64/gcc-9.5.0/intel-oneapi-compilers-2023.1.0-2bw2doei7cntq6xo2qub37m2oewsty47/compiler/2023.1.0/linux/include -I/apps/spacks/2023-05-19/opt/spack/linux-rocky9-x86_64/gcc-9.5.0/intel-oneapi-compilers-2023.1.0-2bw2doei7cntq6xo2qub37m2oewsty47/compiler/2023.1.0/linux/include/sycl -DLAPACK_HAVE_ONEMKL-fsycl -fp-model=precise -c config/onemkl.cc -o config/onemkl.o
^[[31micpx: warning: -Wl,-rpath,/apps/spacks/2023-05-19/opt/spack/linux-rocky9-x86_64/gcc-9.5.0/intel-oneapi-mkl-2023.1.0-ytnolxe64pe2eosx5xgi2hgtrf5flub5/mkl/2023.1.0/lib/intel64: 'linker' input unused [-Wunused-command-line-argument]
In file included from <built-in>:437:
<command line>:9:27: warning: ISO C99 requires whitespace after the macro name [-Wc99-extensions]
#define LAPACK_HAVE_ONEMKL-fsycl 1
                          ^
```

Reply:
Sorry, I missed that.  I had looked at the make.inc, and it looked correct.  
Maybe that it is passing the define via the include/lapack/defines.h rather than on the command line.
I have put the space back, and I just finished a test successfully.
```
# make.inc file
# auto-generated by: configure.py
# host: uan-0002
CXXFLAGS = -std=c++17 -O2 -MMD -Wall -Wno-unused-local-typedefs -Wno-unused-function -fopenmp  -I/soft/restricted/CNDA/updates/2022.12.30.001/oneapi/compiler/trunk-20230201/compiler/linux/include -I/soft/restricted/CNDA/updates/2022.12.30.001/oneapi/compiler/trunk-20230201/compiler/linux/include/sycl -fsycl -fp-model=precise

# see include/lapack/defines.h
# -DLAPACK_FORTRAN_ADD_ -DLAPACK_COMPLEX_RETURN_ARGUMENT -DLAPACK_HAVE_MKL -DLAPACK_HAVE_CBLAS -DLAPACK_HAVE_LAPACK -DLAPACK_VERSION=31001 -DLAPACK_HAVE_XBLAS -DLAPACK_HAVE_LAPACKE -DLAPACK_HAVE_ONEMKL
```

Reply:
Tests pass successfully.
```
env ZE_AFFINITY_MASK=0.0 ./run_tests.py --quick
Thu Jul 13 19:58:52 2023
LAPACK++ version 2023.06.00, id c8c6254
...
All routines passed.
Thu Jul 13 20:02:41 2023
```

Reply:
One possible change for testsweeper.
The icpx compiler has given warnings that  "long long start_, end_, step_;' might be used initialized.
```
diff --git a/testsweeper.cc b/testsweeper.cc
index 2d224a9..191e18b 100644
--- a/testsweeper.cc
+++ b/testsweeper.cc
@@ -230,7 +230,7 @@ int64_t scan_multiplier( const char **pstr )
 ///
 int scan_range( const char** pstr, int64_t* start, int64_t* end, int64_t* step )
 {
-    long long start_, end_, step_;
+    long long start_ = 0, end_ = 0, step_ = 0;
     int bytes;
     const char* str = *pstr;
```

Reply:
[Already fixed](https://github.com/icl-utk-edu/testsweeper/commit/d2491c7d2210cae3f4e234b7421d43145c6df6c7). Update TestSweeper.